packages/base/src/components/files/FilesListing.tsx

```diff
@@ -0,0 +1,219 @@
+import * as React from 'react';
+import { Button } from 'react-bootstrap';
+import { Map, Set, List } from 'immutable';
+import { IFile } from "./models";
+import { FileAttachmentForm } from "./FileAttachmentForm";
+
+interface Props {
+    files: List<IFile>
+    addFileText: string
+    noFilesMessage: string
+
+    handleUpload?: (files: Map<string, File>, cb: () => any) => any
+    handleDelete?: (file: string) => any
+    handleDownload?: (files: Set<string>) => any
+    canInsert?: boolean
+    canDelete?: boolean
+    useFilePropertiesEditTrigger?: boolean
+    onPropertyUpdate?: () => any
+    onSubmit?: () => any
+    onUploadFiles?: () => any
+    getFilePropertiesEditTrigger?: (file :IFile) => React.ReactNode
+}
+
+interface State {
+    selectedFiles?: Set<string>
+    showConfirmDelete: Set<string>
```

Comment:
The naming here is a bit confusing since generally the `showXXX` is a boolean.  I might call it something like `confirmDeletionSet` or `deletionSelection`.